test.js

```diff
@@ -94,36 +83,48 @@ describe('cli', function () {
   describe('with mocked git', function () {
     it('--sign signs the commit and tag', function () {
       // mock git with file that writes args to gitcapture.log
-      mockGit('require("fs").appendFileSync("gitcapture.log", JSON.stringify(process.argv.splice(2)) + "\\n", "utf8")')
-      writePackageJson('1.0.0')
+      mockGit('require("fs").appendFileSync("gitcapture.log", JSON.stringify(process.argv.splice(2)) + "\\n")')
+        .then(function (unmock) {
+          writePackageJson('1.0.0')
+
+          shell.exec(cliPath + ' --sign').code.should.equal(0)
 
-      shell.exec(cliPath + ' --sign').code.should.equal(0)
+          var captured = shell.cat('gitcapture.log').stdout.split('\n').map(function (line) {
+            return line ? JSON.parse(line) : line
+          })
+          captured[captured.length - 3].should.deep.equal(['commit', '-S', 'package.json', 'CHANGELOG.md', '-m', 'chore(release): 1.0.1'])
+          captured[captured.length - 2].should.deep.equal(['tag', '-s', 'v1.0.1', '-m', 'chore(release): 1.0.1'])
 
-      var captured = shell.cat('gitcapture.log').stdout.split('\n').map(function (line) {
-        return line ? JSON.parse(line) : line
-      })
-      captured[captured.length - 3].should.deep.equal(['commit', '-S', 'package.json', 'CHANGELOG.md', '-m', 'chore(release): 1.0.1'])
-      captured[captured.length - 2].should.deep.equal(['tag', '-s', 'v1.0.1', '-m', 'chore(release): 1.0.1'])
+          unmock()
+        })
     })
 
     it('exits with error code if git commit fails', function () {
       // mock git by throwing on attempt to commit
-      mockGit('if (process.argv[2] === "commit") { console.error("commit yourself"); process.exit(128); }')
-      writePackageJson('1.0.0')
+      mockGit('console.error("commit yourself"); process.exit(128);', 'commit')
```

Comment:
This is an example of using `command`. If we did use `git push` here, it shouldn't be mocked @bcoe